src/browser/BrowserService.cpp

```diff
@@ -1572,11 +1572,11 @@ bool BrowserService::handleURLWithWildcards(const QUrl& entryQUrl, const QString
         }
 
         // Escape illegal characters
-        auto re = firstPart.replace(QRegularExpression(R"(([!\^\$\+\-\(\)@<>]))"), "\\\\1");
+        auto re = Tools::escapeRegex(firstPart);
 
         if (hostnameUsed) {
             // Replace all host parts with wildcards
-            re = re.replace(QString("%1.").arg(UrlTools::URL_WILDCARD), "(.*?)");
+            re = re.replace(QString("%1.").arg(UrlTools::URL_WILDCARD), "(.*?)\\.");
```

Comment:
Doing just this breaks the compatibility with the URLs that have now been removed from the tests. If we still want to match the URLs with or without subdomains, one way to do that is to make the first wildcard optional if it only contains the host:
```
if (re.startsWith(QString("%1\\.%2").arg(UrlTools::URL_WILDCARD, Tools::escapeRegex(secondPart)))) {
    re.replace(0, UrlTools::URL_WILDCARD.length()+2, "(?:.*?\\.)?");
}
```

Or something similar. Otherwise URL `https://*.github.com/*` will match `https://login.github.com/path` but not `https://github.com/path`. But if we want to make this less complex, I'm ok with the new restriction.